### PR TITLE
Crook

### PR DESCRIPTION
**Pull request by CreeperFace0777**
## Crook
* 1/4 chance of leaf block dropping a sapling when mined with a crook
* Durability Shown in lore and shown with colour in item display name
* 2 Recipies added to obtain the crook in survival
* No Command necessary

All bugs have been fixed. All merge conflicts have been resolved.
Please check code for optimum efficiency

###For a compiled version of the code, please check release v0.0.1

### CODE REVIEW

File: src/team/crazynetwork/raids/Island.java [hunk 1]
```diff
@@ -10,17 +10,19 @@
 import java.util.UUID;
 
 public class Island {
-    public static List<Island> islands = new ArrayList<>();
     private int x, y, z; //The coordinates of the island
     private Long raidableTime; //When it will be raidable
     private Player owner; //The owner
     private List<Player> members; //Members of the island
+    private double balence; //Island Balance
```
Comment (chanbakjsd) [inline src/team/crazynetwork/raids/Island.java]: Why is "balance" called "balence"? Makes it harder to understand when reading code from bottom without looking at comments.

File: src/team/crazynetwork/raids/Island.java [hunk 2]
```diff
@@ -31,8 +33,7 @@ public Island(Player owner) {
         this.raidableTime = pIsland.getLong("raidableTime");
         for (String member : pIsland.getStringList("members"))
             this.members.add(Bukkit.getPlayer(UUID.fromString(member)));
-
-
+        this.balence = pIsland.getDouble("balence");
```
Comment (chanbakjsd) [inline src/team/crazynetwork/raids/Island.java]: Error checking to be added. Don't always expect config to be full of things. Include default values.

File: src/team/crazynetwork/raids/skyfactory/Crook.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                                            //The item is most held likely a crook
+                                            if (ChatColor.stripColor(e.getPlayer().getItemInHand().getItemMeta().getLore().get(1)).equalsIgnoreCase("1/64")) {
+                                                //The crook is out of durability now
+                                                if (e.getPlayer().getItemInHand().getAmount() == 1) {
+                                                    e.getPlayer().setItemInHand(null); //If there is only 1 crook in their hand, just remove everything from the hand
+                                                } else {
+                                                    ItemStack c = recipe()[0].getResult();
+                                                    c.setAmount(e.getPlayer().getItemInHand().getAmount() - 1);
+                                                    e.getPlayer().setItemInHand(c); //If there is more than 1 crook, delete 1 of them and make the rest full durability
```
Comment (chanbakjsd) [inline src/team/crazynetwork/raids/skyfactory/Crook.java:53]: Maybe times two the durability? If a player decide to stack two durability 1 crooks and break a leaf, he will get a full durability crook. Only items with same lore stack.

File: src/team/crazynetwork/raids/skyfactory/Crook.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                                                e.getPlayer().playSound(e.getPlayer().getLocation(), Sound.ITEM_BREAK, 1F, 1F); //Make the tool break sound
+                                                return;
+                                            } else {
+                                                ItemMeta newMeta = e.getPlayer().getItemInHand().getItemMeta();
+                                                List<String> lore = newMeta.getLore();
+                                                int dur = Integer.parseInt(ChatColor.stripColor(newMeta.getLore().get(1).split("/")[0])) - 1; //New Durability
+                                                lore.set(1, ChatColor.RESET + "" + ChatColor.DARK_GRAY + dur + "/64");
+                                                newMeta.setLore(lore);
+                                                ChatColor coloredDur = ChatColor.DARK_GREEN; //Changed color of the durability depending on how low/high it is
+                                                if (dur < 63) coloredDur = ChatColor.GREEN;
```
Comment (chanbakjsd) [inline src/team/crazynetwork/raids/skyfactory/Crook.java:65]: Use <= instead of <

File: src/team/crazynetwork/raids/skyfactory/Crook.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                                                if (dur < 48) coloredDur = ChatColor.YELLOW;
+                                                if (dur < 32) coloredDur = ChatColor.GOLD;
+                                                if (dur < 16) coloredDur = ChatColor.RED;
+                                                if (dur < 3) coloredDur = ChatColor.DARK_RED;
+                                                newMeta.setDisplayName(ChatColor.RESET + "Crook " + coloredDur + dur + "/64");
+                                                e.getPlayer().getItemInHand().setItemMeta(newMeta);
+                                                //Set the lore and the display name of the crook to the durability
+                                            }
+                                            if (new Random().nextInt(4) == 1) { //just under a 1 in 3 chance of success
```
Comment (chanbakjsd) [inline src/team/crazynetwork/raids/skyfactory/Crook.java]: Change the comment. 1 in 4 chance.

File: src/team/crazynetwork/raids/skyfactory/Crook.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                                                e.setCancelled(true);
+                                                e.getBlock().getWorld().dropItemNaturally(e.getBlock().getLocation(), new ItemStack(Material.SAPLING, 1)); //Set the drop as a sapling
```
Comment (chanbakjsd) [inline src/team/crazynetwork/raids/skyfactory/Crook.java]: Only drop 1 sapling? I thought this code is about dropping MULTIPLE saplings.

File: src/team/crazynetwork/raids/Island.java (hunk 1 above)
Comment (CreeperFace0777) [reply src/team/crazynetwork/raids/Island.java]: Spelling mistake

File: src/team/crazynetwork/raids/skyfactory/Crook.java (hunk 6 above)
Comment (CreeperFace0777) [reply src/team/crazynetwork/raids/skyfactory/Crook.java]: After testing, it was concluded more than one sapling drop was too high.

Comment (chanbakjsd) [reply src/team/crazynetwork/raids/skyfactory/Crook.java]: What's the point of adding crook then? It is meant to increase drop rate.

Comment (CreeperFace0777) [reply src/team/crazynetwork/raids/skyfactory/Crook.java]: Saplings don't drop ~1 in 4 times in vanilla. The comment says 1/3 as it was (at one point) a one in three chance. Still thought it was too high

Comment (chanbakjsd) [reply src/team/crazynetwork/raids/skyfactory/Crook.java]: Right. Move on.

**Review by chanbakjsd — APPROVED** [review]
Perfect. You merge it.

**Pull request merged**